test/integration/dss/test_dss_api.py

```diff
@@ -20,15 +21,41 @@
 
 if USING_PYTHON2:
     import backports.tempfile
+    import mock
     import unittest2 as unittest
     TemporaryDirectory = backports.tempfile.TemporaryDirectory
 else:
     import unittest
+    from unittest import mock
     TemporaryDirectory = tempfile.TemporaryDirectory
 
+
 class TestDssApi(unittest.TestCase):
     staging_bucket = "org-humancellatlas-dss-cli-test"
 
+    def test_set_host(self):
+        with TemporaryDirectory() as home:
+            with mock.patch.dict(os.environ, HOME=home):
+                dev = hca.dss.DSSClient(
+                    swagger_url="https://dss.dev.data.humancellatlas.org/v1/swagger.json")
+                self.assertEqual("dss.dev.data.humancellatlas.org", dev._swagger_spec['host'])
+
+    def test_set_host_multithreaded(self):
```

Comment:
Do these two tests cover all the cases I brought up in the ticket?

Reply:
I would say so. These tests cover:
- specifying a "host" via `swagger_url` on client initialization
- ensuring that the host is set appropriately for a given `swagger_url` and does not revert to prod when multiple clients are initialized across multiple threads

Redefining the client host after initialization would require undoing and redoing the code generation of the API bindings for which we could just create a new client instead. The changes also no longer require the developer to use the HCAConfig workaround.

Reply:
Ok. Just to be sure, I will run my test suite again, minus the workarounds.